ReactWindows/ReactNative.Tests/Modules/Clipboard/ClipboardModuleTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
+using ReactNative.Bridge;
+using ReactNative.Modules.Clipboard;
+using System;
+
+namespace ReactNative.Tests.Modules.Clipboard
+{
+    [TestClass]
+    public class ClipboardModuleTests
+    {
+        [TestMethod]
+        public void ClipboardModule_ArgumentChecks()
+        {
+            var module = new ClipboardModule(new ReactContext());
+
+            AssertEx.Throws<ArgumentNullException>(
+                () => module.getString(null),
+                ex => Assert.AreEqual("promise", ex.ParamName));
+        }
+
+        [TestMethod]
+        public void ClipboardModule_GetString_Method()
+        {
+            if (!ReactNative.Bridge.DispatcherHelpers.IsInitialized) return;
+
+            var module = new ClipboardModule(new ReactContext());
+            var str = "test string";
+
+            var promise = new MockPromise(resolve => Assert.AreEqual(str, resolve.ToString()), reject => str = reject);
+
+            module.setString(str);
+            module.getString(promise);
+        }
+
+        [TestMethod]
+        public void ClipboardModule_SetString_Null_Method()
+        {
+            if (!ReactNative.Bridge.DispatcherHelpers.IsInitialized) return;

```

Comment:
If you use Context.RunOnDispatcherQueueThread in the module, then the best way is to initialize a proper ReactInstance (which will new up all of its queue threads).  Take a look at ReactInstanceTests.cs.  You may be able to use the [UITestMethod] attribute to avoid having to use the Test DispatcherHelpers.

---
In reply to: [55509705](https://github.com/CatalystCode/react-native/pull/233#discussion_r55509705) [](ancestors = 55509705)